core/coreunix/add.go

```diff
@@ -321,42 +266,92 @@ func (adder *Adder) AddAllAndPin(file files.Node) (ipld.Node, error) {
 		}
 	}()
 
-	switch tf := file.(type) {
-	case files.Directory:
-		// Iterate over each top-level file and add individually. Otherwise the
-		// single files.File f is treated as a directory, affecting hidden file
-		// semantics.
-		it := tf.Entries()
-		for it.Next() {
-			if err := adder.addFileNode(it.Name(), it.Node()); err != nil {
-				return nil, err
-			}
+	if err := adder.addFileNode("", file, true); err != nil {
+		return nil, err
+	}
+
+	// get root
+	mr, err := adder.mfsRoot()
+	if err != nil {
+		return nil, err
+	}
+	var root mfs.FSNode
+	rootdir := mr.GetDirectory()
+	root = rootdir
+
+	err = root.Flush()
+	if err != nil {
+		return nil, err
+	}
+
+	// if adding a file without wrapping, swap the root to it (when adding a
+	// directory, mfs root is the directory)
+	_, dir := file.(files.Directory)
```

Comment:
I'm having trouble understanding this comment. Why do we care if the file we're adding is a directory?

Reply:
It made sense before I rewrote this code.